src/python/clawutil/runclaw.py

```diff
@@ -204,29 +214,28 @@ def runclaw(xclawcmd=None, outdir=None, overwrite=True, restart=None,
         print("\n==> Running with command:\n   ", cmd)
 
     cmd_split = shlex.split(cmd)
-    if isinstance(xclawout,str):
+    if isinstance(xclawout, str):
         xclawout = open(xclawout,'w', encoding='utf-8',
                         buffering=1)
-    if isinstance(xclawerr,str):
+    if isinstance(xclawerr, str):
         xclawerr = open(xclawerr,'w', encoding='utf-8',
                         buffering=1)
     try:
-        p = subprocess.run(cmd_split,
-                           cwd=outdir,
-                           stdout=xclawout,
-                           stderr=xclawerr,
-                           encoding='utf-8',
-                           check=True)
+        proc = subprocess.check_call(cmd_split,
+                                     cwd=outdir,
+                                     stdout=xclawout,
+                                     stderr=xclawerr)
+
     except subprocess.CalledProcessError as cpe:
         raise ClawExeError('error', cpe.returncode, cpe.cmd,
                            output=cpe.output, 
-                           stderr=cpe.stderr) from cpe
```

Comment:
I think the `from cpe` just keeps the existing call stack when raising the error, so potentially helps a bit with debugging. Is that a python 3-only syntax? Or is there another reason you dropped that?

Reply:
As far as I know this was added to Python 3.x unfortunately.